### PR TITLE
add distanceExcercise

### PR DESCRIPTION
**Pull request by tomijava**

### CODE REVIEW

File: Distance.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import java.util.Objects;
+
+public class Distance {
+
+            /* Napisz klase ktora trzyma informacje o wartosci odleglosci i
```
Comment (tomaszte) [inline Distance.java]: fix intendation lines 5-70

File: Distance.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                }
+            }
+
+
+    public boolean equals(Distance obj) {
+        return this.cmMeasure== obj.cmMeasure;
+    }
+
+    @Override
+        public String toString()
```
Comment (tomaszte) [inline Distance.java]: fix intendation

File: Distance.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                {
+                    cmMeasure = this.distance;
+                }
+                else if(Objects.equals(measure, DistanceUnitType.METERS))
+                {
+                    cmMeasure = this.distance * 100;
+                }
+            }
+
+            public void changerMesure(DistanceUnitType convert) {
```
Comment (tomaszte) [inline Distance.java]: fix method name, and parameter name

File: Distance.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            enum DistanceUnitType
+            {
+                CENTIMETERS,
+                METERS,
+                KILOMETERS
+            }
+
+            private double distance;
+            private DistanceUnitType measure;
+            private double cmMeasure;
```
Comment (tomaszte) [inline Distance.java]: cmMeasure field is redundant i think and we should avoid defining fields that we don't really need.
You could try to implement method 
int getDistanceInUnit(Distance distance, DistanceUnitType unitType);
that will return distance in different unit, but will not modify the object.
You could use that method for comparing distances and changing unit type.

**Review by tomaszte — CHANGES_REQUESTED** [review]

File: Distance.java (hunk 1 above)
Comment (tomijava) [reply Distance.java]: ok :(

File: Distance.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    private double cmMeasure;
+
+    Distance(double distance, DistanceUnitType measure)
+    {
+        this.distance = distance;
+        this.measure = measure;
+
+        if(Objects.equals(measure, DistanceUnitType.KILOMETERS))
+        {
+            cmMeasure = this.distance *100000;
```
Comment (tomaszte) [inline Distance.java]: instead of magic number 100000 define constant named like CENTIMETERS_IN_KILOMETER

File: Distance.java [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+public class Distance {
+	/* Napisz klase ktora trzyma informacje o wartosci odleglosci i
+    o rodzaju miary METRY KILOMETRY CENTYMETRY w klasie ma byc metoda ktora
+    porownuje dwa miary medota toString ktora ladnie wypsuize miare np. 10 km i
+    metoda ktora przyjmuje typ miary i zmieni typ miary obiektu na inny
+    *///wtf
+
+
+	enum DistanceUnitType
```
Comment (tomaszte) [inline Distance.java]: spacing

File: Distance.java [hunk 7]
```diff
@@ -0,0 +1,101 @@
+import java.util.Objects;
+
+public class Distance {
+	/* Napisz klase ktora trzyma informacje o wartosci odleglosci i
```
Comment (tomaszte) [inline Distance.java]: spacing

File: Distance.java [hunk 8]
```diff
@@ -0,0 +1,101 @@
+import java.util.Objects;
+
+public class Distance {
+	/* Napisz klase ktora trzyma informacje o wartosci odleglosci i
+    o rodzaju miary METRY KILOMETRY CENTYMETRY w klasie ma byc metoda ktora
+    porownuje dwa miary medota toString ktora ladnie wypsuize miare np. 10 km i
+    metoda ktora przyjmuje typ miary i zmieni typ miary obiektu na inny
+    *///wtf
```
Comment (tomaszte) [inline Distance.java]: remove that ///wtf 💃

File: Distance.java (hunk 5 above)
Comment (tomaszte) [reply Distance.java]: do that for every 'magic number'
https://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad

File: Distance.java [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+            {
+               this.distance /= 1000;
+               this.measure = measure;
+            }
+    	} 
+    	else if (Objects.equals(measure, DistanceUnitType.METERS)) 
+    	{
+        	if (Objects.equals(this.measure, DistanceUnitType.KILOMETERS)) 
+        	{
+            	this.distance *= 1000;
```
Comment (tomaszte) [inline Distance.java]: spacing

**Review by tomaszte — CHANGES_REQUESTED** [review]

**Pull request merged**